tests/integration/test_source_git_generator.py

```diff
@@ -0,0 +1,144 @@
+# Copyright Contributors to the Packit project.
+# SPDX-License-Identifier: MIT
+import subprocess
+from pathlib import Path
+
+import pytest
+
+from packit.fedpkg import FedPKG
+from packit.local_project import LocalProject
+from packit.patches import PatchMetadata
+from packit.source_git import SourceGitGenerator
+from packit.specfile import Specfile
+from tests.spellbook import initiate_git_repo
+
+UNIVERSAL_PACKAGE_NAME = "redhat-rpm-config"
+
+
+@pytest.mark.parametrize(
+    "fedora_package, centos_package, branch",
+    (
+        (UNIVERSAL_PACKAGE_NAME, None, "master"),
+        (None, UNIVERSAL_PACKAGE_NAME, "c8s"),
+    ),
+)
+def test_distgit_cloning(
+    api_instance_source_git, fedora_package, centos_package, branch, tmp_path: Path
+):
+    sgg = SourceGitGenerator(
+        api_instance_source_git.upstream_local_project,
+        api_instance_source_git.config,
+        # yes, this is the upstream repo
+        f"https://src.fedoraproject.org/rpms/{UNIVERSAL_PACKAGE_NAME}",
+        fedora_package=fedora_package,
+        centos_package=centos_package,
+        dist_git_branch=branch,
+        tmpdir=tmp_path,
+    )
+    sgg._get_dist_git()
+    assert tmp_path.joinpath(UNIVERSAL_PACKAGE_NAME, ".git").is_dir()
+
+
+# TODO: test different refs: tag, branch, commit hash
+def test_fetch_upstream_ref(api_instance_source_git, tmp_path: Path):
+    tag = "1.0.0"
+    s = tmp_path.joinpath("s")
+    u = tmp_path.joinpath("u")
+    u.mkdir()
+    subprocess.check_call(["git", "init", str(s)])
+    initiate_git_repo(u, tag=tag)
+    sgg = SourceGitGenerator(
+        LocalProject(working_dir=s),
+        api_instance_source_git.config,
+        str(u),
+        upstream_ref=tag,
+        centos_package="x",
+    )
+
+    sgg._pull_upstream_ref()
+
+    assert s.joinpath(".git").is_dir()
+    assert sgg.local_project.ref == "master"
+    assert sgg.local_project.working_dir.joinpath("hops").read_text() == "Cascade\n"
+    assert sgg.local_project.git_repo.head.commit.message == "commit with data\n"
+
+
+@pytest.mark.parametrize(
+    "fedora_package, centos_package, branch",
+    (("fuse-overlayfs", None, "master"),),
+)
+def test_run_prep(
+    api_instance_source_git, fedora_package, centos_package, branch, tmp_path: Path
+):
+    sgg = SourceGitGenerator(
+        api_instance_source_git.upstream_local_project,
+        api_instance_source_git.config,
+        f"https://github.com/containers/{fedora_package}",
+        fedora_package=fedora_package,
+        centos_package=centos_package,
+        dist_git_branch=branch,
+        tmpdir=tmp_path,
+    )
+    assert sgg.primary_archive.exists()  # making sure this is downloaded
+    sgg._run_prep()
+    B = sgg.dist_git.local_project.working_dir.joinpath("BUILD")
+    assert B.exists()
+    project_dir = next(B.glob("fuse-overlayfs-*"))
```

Comment:
I don't like one-letter variables in general (like above..;( , but the upper-case one-letter variable is something new here...

```suggestion
    build_dir = sgg.dist_git.local_project.working_dir.joinpath("BUILD")
    assert build_dir.exists()
    project_dir = next(build_dir.glob("fuse-overlayfs-*"))
```

Reply:
because it's for the ./BUILD/{package}-{version} directory, hence capital "B"